src/utils/utils_load_library.h

```diff
@@ -32,29 +36,42 @@ extern "C" {
 
 #ifdef _WIN32 /* Windows */
 
-static inline void *util_open_library(const char *filename) {
+static inline void *util_open_library(const char *filename, int dlopenFlags) {
+    (void)dlopenFlags; //unused
     return LoadLibrary(TEXT(filename));
 }
 
 static inline int util_close_library(void *handle) {
     // If the FreeLibrary function succeeds, the return value is nonzero.
     // If the FreeLibrary function fails, the return value is zero.
-    return (FreeLibrary(handle) == 0);
+    return (FreeLibrary((HMODULE)handle) == 0);
 }
 
-static inline void *util_get_symbol_addr(void *handle, const char *symbol) {
-    return GetProcAddress(handle, symbol);
+static inline void *util_get_symbol_addr(void *handle, const char *symbol,
+                                         const char *libname) {
+    if (!handle) {
+        if (libname == NULL) {
+            return NULL;
+        }
+        handle = GetModuleHandle(libname);
+    }
+    return GetProcAddress((HMODULE)handle, symbol);
 }
 
 #else /* Linux */
 
-static inline void *util_open_library(const char *filename) {
-    return dlopen(filename, RTLD_LAZY);
+static inline void *util_open_library(const char *filename, int dlopenFlags) {
```

Comment:
I know i will be annoying - but if you create cross-platform utility which accepts flags - you have to define those flags by your self, and translate them inside of the function to OS specific one.